Tailwind redesign + new Search (hidden)

msafi opened the pull request.

levsthings commented:
LGTM but this needs to be in wiki as a future reference. We'll definitely remember this but in case someone else is delegated to do this update in the future, they need to be linked these instructions.

msafih commented:
The greenkeeper PR causes a conflict with the `yarn.lock` file, so I resolved the conflict by merging master into this PR and regenerating the `yarn.lock` file, but I think what I did brought in the new typing package or something that broke our build. We need to resolve that before I can merge this PR

msafih commented:
And actually, Travis CI is broken with this PR for some reason. I'll try to close it, delete it, and push again.

rlogank committed:
init

rlogank committed:
restyled homepage with tailwindcss, responsive

rlogank committed:
fixed minor issues

msafi committed:
Add react debounce input

msafi committed:
Add new search scaffolding behind feature flag

rlogank committed:
celeb page redesign part 1

rlogank committed:
added a lovely top border

msafi committed:
Create kgCall

rlogank committed:
added some fixes and tweaks

rlogank committed:
updated font + couple fixes

rlogank committed:
welcome

msafi committed:
Add prettier and undo unintended prettier formatting changes

msafi committed:
before trying search route

msafi committed:
initial search 100% complete

msafi committed:
Knowledge Graph Celeb page 70% done

msafi committed:
Log missing celebs to Google Analytics

rlogank committed:
Removed SCSS and MUI, adapted all affected components, added accessibility features

msafi committed:
Merge branch 'search' into tw-redesign

# Conflicts:
#	lib/[celeb]/Facts/Fact/styles.module.scss
#	lib/[celeb]/index.tsx
#	lib/_app/AppBar/AppBar.tsx
#	lib/_app/index.tsx
#	lib/index/index.tsx
#	lib/~/contribute/index.tsx

msafi committed:
Fix some merge issues

rlogank committed:
added footer section, modified a blockquote

msafi committed:
Make Logan's signature higher contrast and reduce click surface

msafi committed:
use isEmpty to check arrays on celeb page

msafi committed:
Change `div` to `a` because that's more semantic. Using `div` was also breaking the CMD+click to open in new window feature.

Also, change the text from HOLLOWVERSE to Hollowverse then add `uppercase` class to achieve the visual effect because that's more semantically correct.

msafi committed:
Add recommended extensions so that VS Code users will get Extension recommendations when they open this project in VS Code.

msafi committed:
Trim bloat caused by empty `className`

msafi committed:
Create a reusable Card component to avoid repeating code

msafi committed:
The styles that are in globalStyles.css are mostly meant for the Editorial, but the way they were written they were affecting the entire app. I moved them to a CSS module and scoped them to only the Editorial section.

msafi committed:
Create a reusable Tag component

msafi committed:
Create a reusable lovely top border

msafi committed:
Clean up CelebGallery

msafi committed:
Use the Card component to display Facts

msafi committed:
Remove separator component

msafi committed:
Improve markup semantics on the contribute page

msafi committed:
Delete unused Icon component

msafi committed:
Miscellaneous cleanup

msafi committed:
Improve markup semantics in Fact

msafi committed:
Miscellaneous cleanup

msafi committed:
Reduce app width

msafi committed:
Refactor directory structure in lib/

msafi committed:
Refactor api/ and ~contribute

msafi committed:
Override default `lg` breakpoint and...

Set it to `896px` to match `max-w-4xl`

msafi committed:
Add Google Search

msafi committed:
Fix style issue

msafi committed:
Fix AppBar

msafi committed:
Use next/link on Homepage

rlogank committed:
Completed pre-merge todo list

msafi committed:
Fix breakpoint setting

msafi committed:
Clean up card component

msafi committed:
Clean up classes in TopSection

msafi committed:
Optimize images

msafi committed:
Add accessibility linting

msafi committed:
Clean up unnecessary classes

msafi committed:
Reverse some removed classes
